graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -141,21 +181,41 @@
 class TypeRenamingMapping(Protocol):
     def get(self, key: str, default: Optional[str]) -> Optional[str]:
         """Define mapping for type_renamings object."""
-        ...
+
+
+# Unfortunately this can't be a nested class because mypy has a bug and it's safer to make
+# the classes slightly less organized in exchange for being able to use mypy here.
+# https://github.com/python/mypy/issues/6393
+class FieldRenamingsForType(Protocol):
```

Comment:
I'd like this to work with dicts, but that would require the default argument and the return to be Optional[Set[str]]. However, it's natural to represent suppressions as the empty set, so actually changing the type hint here to be Optional would be misleading.